tests/unittest/inputs/test_prefix_token_cache.py

```
@@ -0,0 +1,176 @@
+def _cache(**kwargs):
+    defaults = dict(max_entries=64, overlap=4, resync=2, min_chars=0, bucket_chars=16)
+    defaults.update(kwargs)
+    return PrefixTokenCache(**defaults)
+
+
+def _turns(n=12):
+    """Prompts that grow by a small delta, as multi-turn serving does."""
+    text = PREAMBLE
+    out = []
+    for i in range(n):
+        text += f"turn {i}: theابcd inner answer aber {i} thereabouts. "
```

Ruff (0.16.2)

[warning] String contains ambiguous `ا` (ARABIC LETTER ALEF). Did you mean `l` (LATIN SMALL LETTER L)?

(RUF001)